### PR TITLE
feat: consensus support for pruning (BFT-473)

### PR DESCRIPTION
**Pull request by pompon0**
Added a task which is monitoring the persisted block range and notifies consensus if it shrinks.

### CODE REVIEW

File: core/node/consensus/src/storage/mod.rs [hunk 1]
```diff
@@ -440,23 +387,75 @@ impl Store {
 
 impl StoreRunner {
     pub async fn run(mut self, ctx: &ctx::Ctx) -> anyhow::Result<()> {
-        let res = async {
+        let res = scope::run!(ctx, |ctx, s| async {
```
Comment (popzxc) [inline core/node/consensus/src/storage/mod.rs:390]: Let's refactor it as suggested in DMs.

File: core/lib/dal/src/consensus_dal.rs [hunk 2]
```diff
@@ -196,68 +189,103 @@ impl ConsensusDal<'_, '_> {
...
+        .report_latency()
+        .fetch_optional(&mut txn)
+        .await?;
+        txn.commit().await.context("commit()")?;
+        Ok(BlockStoreState {
+            first: start,
+            last: match row {
+                None => None,
+                Some(row) => Some(zksync_protobuf::serde::deserialize(row.certificate)?),
+            },
```
[the remark this answers is not in the document]
Comment (pompon0) [reply core/lib/dal/src/consensus_dal.rs:279]: done, I don't like transpose() though

File: core/lib/dal/src/consensus_dal.rs [hunk 3]
```diff
@@ -196,68 +189,103 @@ impl ConsensusDal<'_, '_> {
...
+            .context("get_sealed_l2_block_number()")?
+        {
+            return Ok(validator::BlockNumber(last.0.into()) + 1);
+        }
+        let next = txn
+            .consensus_dal()
+            .first_block()
+            .await
+            .context("first_block()")?;
+        txn.commit().await.context("commit()")?;
```
[the remark this answers is not in the document]
Comment (pompon0) [reply core/lib/dal/src/consensus_dal.rs:232]: I don't know if it is needed yet, I've preemptively added it everywhere. I missed that one place though.

File: core/node/consensus/src/storage/mod.rs (hunk 1 above)
Comment (pompon0) [reply core/node/consensus/src/storage/mod.rs:390]: done

File: core/lib/dal/src/consensus_dal.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (pompon0) [reply core/lib/dal/src/consensus_dal.rs:232]: I've removed the commits for now, until the question whether it is needed or not is resolved.

File: Cargo.lock [hunk 4]
```diff
@@ -2999,7 +3117,7 @@ dependencies = [
  "beef",
  "futures-timer",
  "futures-util",
- "hyper",
+ "hyper 0.14.29",
```
[the remark this answers is not in the document]
Comment (pompon0) [reply Cargo.lock:3120]: ack

**Review by popzxc — DISMISSED** [review]

**Review by popzxc — APPROVED** [review]

**Pull request merged**